pkg/declarativeconfig/transform/role_test.go

```diff
@@ -0,0 +1,48 @@
+package transform
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWrongConfigurationTypeTransform(t *testing.T) {
+	rt := newRoleTransform()
+	protos, err := rt.Transform(&declarativeconfig.AuthProvider{})
+	assert.Nil(t, protos)
+	assert.Error(t, err)
+	assert.ErrorIs(t, err, errox.InvalidArgs)
+}
+
+func TestTransform(t *testing.T) {
+	role := &declarativeconfig.Role{
+		Name:          "some-role",
+		Description:   "with a nice description",
+		AccessScope:   "and an access scope",
+		PermissionSet: "as well as a permission set",
```

Comment:
I love the storytelling approach to writing test data :)

Reply:
I got bored of just using `test values`, mabye next time I'll try to make an even better story out of it 😛